infra/config-sync/src/main/java/com/simprints/infra/config/sync/usecase/HandleProjectStateUseCase.kt

```diff
@@ -1,31 +1,18 @@
 package com.simprints.infra.config.sync.usecase
 
-import com.simprints.infra.authlogic.AuthManager
-import com.simprints.infra.authlogic.worker.SecurityStateScheduler
 import com.simprints.infra.config.store.models.ProjectState
-import com.simprints.infra.config.sync.ProjectConfigurationScheduler
 import com.simprints.infra.eventsync.EventSyncManager
-import com.simprints.infra.images.ImageUpSyncScheduler
 import kotlinx.coroutines.flow.first
 import javax.inject.Inject
 
 internal class HandleProjectStateUseCase @Inject constructor(
-    private val configScheduler: ProjectConfigurationScheduler,
-    private val securityStateScheduler: SecurityStateScheduler,
-    private val imageUpSyncScheduler: ImageUpSyncScheduler,
     private val eventSyncManager: EventSyncManager,
-    private val authManager: AuthManager
+    private val logoutUseCase: LogoutUseCase,
 ) {
 
     suspend operator fun invoke(projectId: String, state: ProjectState) {
         if (shouldSignOut(projectId, state)) {
```

Comment:
If the project is `ENDED`, you should try to sync the record and then log out even if the sync fails